### ISSUE TITLE
git dependencies aren't working

### ISSUE THREAD

**Issue by radix** [sdispater/poetry#49]
It seems poetry is passing a PosixPath to execv, which needs strings. Maybe this is something that only affects Python 2.7?

**Version**: poetry 0.8.3

I added a print statement to see what git command was being run, that's the `RUNNING GIT:` part in the output.

```
{}radix@pop-os:~/Juice/fruition$ poetry install -v
Using virtualenv: /home/radix/.cache/pypoetry/virtualenvs/fruition-py2.7
Updating dependencies
Resolving dependencies
RUNNING GIT: ('clone', u'git+https://github.com/imgix/imgix-python', PosixPath('/tmp/pypoetry-git-imgix0ivPX4'))


  [TypeError]
  execv() arg 2 must contain only strings


Exception trace:
 /home/radix/.local/lib/python2.7/site-packages/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /home/radix/.local/lib/python2.7/site-packages/poetry/console/application.py in do_run() at line 74
   return super(Application, self).do_run(i, o)
 /home/radix/.local/lib/python2.7/site-packages/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /home/radix/.local/lib/python2.7/site-packages/poetry/console/commands/command.py in run() at line 22
   return super(BaseCommand, self).run(i, o)
 /home/radix/.local/lib/python2.7/site-packages/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /home/radix/.local/lib/python2.7/site-packages/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /home/radix/.local/lib/python2.7/site-packages/poetry/console/commands/install.py in handle() at line 40
   return installer.run()
 /home/radix/.local/lib/python2.7/site-packages/poetry/installation/installer.py in run() at line 73
   self._do_install(local_repo)
 /home/radix/.local/lib/python2.7/site-packages/poetry/installation/installer.py in _do_install() at line 171
   ops = solver.solve(request, fixed=fixed)
 /home/radix/.local/lib/python2.7/site-packages/poetry/puzzle/solver.py in solve() at line 41
   graph = resolver.resolve(requested, base=base)
 /home/radix/.local/lib/python2.7/site-packages/poetry/mixology/resolver.py in resolve() at line 39
   base
 /home/radix/.local/lib/python2.7/site-packages/poetry/mixology/resolution.py in resolve() at line 72
   self._start()
 /home/radix/.local/lib/python2.7/site-packages/poetry/mixology/resolution.py in _start() at line 113
   self._handle_missing_or_push_dependency_state(self._initial_state())
 /home/radix/.local/lib/python2.7/site-packages/poetry/mixology/resolution.py in _initial_state() at line 220
   self._possibilities_for_requirement(initial_requirement, graph),
 /home/radix/.local/lib/python2.7/site-packages/poetry/mixology/resolution.py in _possibilities_for_requirement() at line 908
   self._provider.search_for(requirement)
 /home/radix/.local/lib/python2.7/site-packages/poetry/puzzle/provider.py in search_for() at line 75
   packages = self.search_for_vcs(dependency)
 /home/radix/.local/lib/python2.7/site-packages/poetry/puzzle/provider.py in search_for_vcs() at line 115
   git.clone(dependency.source, tmp_dir)
 /home/radix/.local/lib/python2.7/site-packages/poetry/vcs/git.py in clone() at line 42
   return self.run('clone', repository, dest)
 /home/radix/.local/lib/python2.7/site-packages/poetry/vcs/git.py in run() at line 100
   stderr=subprocess.STDOUT
 /usr/lib/python2.7/subprocess.py in check_output() at line 212
   process = Popen(stdout=PIPE, *popenargs, **kwargs)
 /usr/lib/python2.7/subprocess.py in __init__() at line 390
   errread, errwrite)
 /usr/lib/python2.7/subprocess.py in _execute_child() at line 1025
   raise child_exception

install [--no-dev] [--dry-run] [-E|--extras EXTRAS]
```

**Comment by sdispater** [comment]
Yes, it seems it slipped through when I ported the code to Python 2.7.

I will fix it.

**Comment by sdispater** [comment]
It should be fixed in the latest release (`0.8.4`).

Update `poetry` with `poetry self:update`.

**Comment by radix** [comment]
Thanks @sdispater ! The new version works. However, I had to use pip to upgrade instead of poetry, because of https://github.com/sdispater/poetry/issues/53
